Implement AWS Security Hub Scan Parser

xee5ch opened the pull request:
Implements requested functionality in #2167.

This template is for your information. Please clear everything when submitting your pull request.

**Note: DefectDojo is now on Python3.5 and Django 2.2.x Please submit your pull requests to the 'dev' branch as the 'legacy-python2.7' branch is only for bug fixes. Any new features submitted to the legacy branch will be ignored and closed.**

When submitting a pull request, please make sure you have completed the following checklist:

- [x] Your code is flake8 compliant.
- [x] Your code is python 3.5 compliant (specific python >=3.6 syntax is currently not accepted).
- [x] If this is a new feature and not a bug fix, you've included the proper documentation in the ReadTheDocs documentation folder. https://github.com/DefectDojo/Documentation/tree/master/docs or provide feature documentation in the PR.
- [x] Model changes must include the necessary migrations in the dojo/db_migrations folder.
- [x] Add applicable tests to the unit tests.
- [x] Add the proper label to categorize your PR.


Current accepted labels for PRs:
- Import Scans (for new scanners/importers)
- enhancement
- performance
- feature
- bugfix
- maintenance (a.k.a chores)
- dependencies
- New Migration

# Git Tips
## Rebase on dev branch
If the dev branch has changed since you started working on it, please rebase your work after the current dev.

On your working branch `mybranch`:
```
git rebase dev mybranch
```
In case of conflict:
```
 git mergetool
 git rebase --continue
 ```

When everything's fine on your local branch, force push to your `myOrigin` remote: 
```
git push myOrigin --force-with-lease
```

To cancel everything: 
```
git rebase --abort
```


## Squashing commits
```
git rebase -i origin/dev
```
- Replace `pick` by `fixup` on the commits you want squashed out
- Replace `pick` by `reword` on the first commit if you want to change the commit message
- Save the file and quit your editor

Force push to your `myOrigin` remote: 
```
git push myOrigin --force-with-lease
```

Maffooch commented:
Thank you for the PR, but could you base it against dev as mentioned in the instructions?

xee5ch commented:
> Thank you for the PR, but could you base it against dev as mentioned in the instructions?

@Maffooch, sorry about that. I pointed it to dev, but I will clean up the `master` merge finish cleanup soon. I did not take off draft status yet because I have not completed tests and other requirements anyway.

Thanks again!

Maffooch commented:
Take as much time as you need! Everything looks good so far. If you wouldn't mind, also add the new scanner to the list on the [docs](https://github.com/DefectDojo/Documentation/blob/master/docs/integrations.rst) please. Thanks.

xee5ch commented:
> as mentioned in the instructions?

Can I also get this PR properly labeled? Since I am a new contributor, it would seem I cannot label an existing PR, @Maffooch? Thanks!

madchap commented on dojo/fixtures/test_type.json:
```diff
@@ -606,18 +606,25 @@
+		  "name": "AWS Security Hub Scan"
+		},
+		"model": "dojo.test_type",
+		"pk": 181
```
Will require update, as other importers got merged in between.

xee5ch replied:
Updated.

xee5ch commented:
Ok, @Maffooch and @madchap , I fixed up tests, adjusted primary keys as I missed others changes, and added docs elsewhere in the issue. I think this is ready for review.
